### ISSUE TITLE
Special characters are escaped in headings when they shouldn't be

### ISSUE THREAD

**Issue by SethFalco** [vector-im/element-android#6699]
### Steps to reproduce

1. Have a chat open
2. Have the other party send one of the following messages if they're on Element, if not on Element then a message that produces the same source:

<details>
<summary><code># <(^-^<)</code></summary>

```json
{
  "type": "m.room.message",
  "content": {
    "org.matrix.msc1767.message": [
      {
        "body": "# <(^-^<)",
        "mimetype": "text/plain"
      },
      {
        "body": "<h1>&lt;(^-^&lt;)</h1>\n",
        "mimetype": "text/html"
      }
    ],
    "body": "# <(^-^<)",
    "msgtype": "m.text",
    "format": "org.matrix.custom.html",
    "formatted_body": "<h1>&lt;(^-^&lt;)</h1>\n"
  }
}
```

</details>

<details>
<summary><code>/html &lt;h1&gt;&amp;lt&semi;(^-^&amp;lt&semi;)&lt;/h1&gt;</code></summary>

```json
{
  "type": "m.room.message",
  "content": {
    "org.matrix.msc1767.message": [
      {
        "body": "<h1>&lt;(^-^&lt;)</h1>",
        "mimetype": "text/plain"
      },
      {
        "body": "<h1>&lt;(^-^&lt;)</h1>",
        "mimetype": "text/html"
      }
    ],
    "body": "<h1>&lt;(^-^&lt;)</h1>",
    "msgtype": "m.text",
    "format": "org.matrix.custom.html",
    "formatted_body": "<h1>&lt;(^-^&lt;)</h1>"
  }
}
```

</details>

#### What did you expect?

Android devices should unescape special characters first before rendering the heading.
For example, on Element Desktop and on GitHub it'll look like this:

# <(^-^<)

#### What happened instead?

The less-than (`<`) symbols aren't unescaped, so the `&lt;` from when the content was converted to HTML is used literally and the user actually sees this:

# \&lt;(^-^\&lt;)

A screenshot from the Android device:
![image](https://user-images.githubusercontent.com/22801583/182044899-f9a2bb3c-f4e8-4682-b988-44c21838d7ab.png)

### Your phone model

Fairphone 4

### Operating system version

Android

### Homeserver

one.ems.host

### Will you send logs?

No

### Are you willing to provide a PR?

Yes

**Comment by paulvt** [comment]
I am not sure what you mean with "headings" but it happens in normal messages too.

Just send some text from Element Android with an ampersand in it. It will appear as `&amp;` when rendered in the timeline.

**Comment by SethFalco** [comment]
Ahh, wasn't ware of that.
I don't use Android so I only know about the issue that was screenshotted and shared with me.

Headings refers to using Markdown to create a heading.

```md
# This is a Heading!
```

# This is a Heading!

**Comment by tulir** [comment]
Duplicate of #6445

**Comment by SethFalco** [comment]
Hmm, I'll make my wife check for updates and and check this out again.
If the issue stops, I'll close the issue.

**Comment by ouchadam** [comment]
thanks for raising! this is indeed #6445 and was fixed as part of [1.4.28](https://github.com/vector-im/element-android/releases/tag/v1.4.28)
